src/nvim/window.c

```diff
@@ -1014,13 +1014,23 @@ void ui_ext_win_position(win_T *wp, bool validate)
       }
       wp->w_winrow = comp_row;
       wp->w_wincol = comp_col;
-      ui_comp_put_grid(&wp->w_grid_alloc, comp_row, comp_col,
-                       wp->w_height_outer, wp->w_width_outer, valid, false);
-      ui_check_cursor_grid(wp->w_grid_alloc.handle);
-      wp->w_grid_alloc.focusable = wp->w_float_config.focusable;
-      if (!valid) {
-        wp->w_grid_alloc.valid = false;
-        redraw_later(wp, UPD_NOT_VALID);
+
+      if (!c.hide) {
+        wp->w_float_flags &= ~FLOAT_HIDDEN;
+        ui_comp_put_grid(&wp->w_grid_alloc, comp_row, comp_col,
+                         wp->w_height_outer, wp->w_width_outer, valid, false);
+        ui_check_cursor_grid(wp->w_grid_alloc.handle);
+        wp->w_grid_alloc.focusable = wp->w_float_config.focusable;
+        if (!valid) {
+          wp->w_grid_alloc.valid = false;
+          redraw_later(wp, UPD_NOT_VALID);
+        }
+      } else {
+        if ((wp->w_float_flags & FLOAT_HIDDEN) != 0) {
+          return;
+        }
+        wp->w_float_flags |= FLOAT_HIDDEN;
```

Comment:
`ui_comp_remove_grid()` is safe to call even if grid wasn't present. Therefore I don't think we need this check, and thus `w_float_flags` field doesn't need to be added at all.